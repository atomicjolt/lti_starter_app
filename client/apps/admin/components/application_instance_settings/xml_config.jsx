import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import _ from 'lodash';
import * as ApplicationInstanceActions from '../../actions/application_instances';
import Textarea from '../common/textarea';

const select = (state, props) => (
  {
    loading: state.applicationInstances.loading,
    loaded: state.applicationInstances.loaded,
    applicationInstances: _.filter(
      state.applicationInstances.applicationInstances,
      { application_id: parseInt(props.params.applicationId, 10) }
    ),
  }
);

export function XmlConfig(props) {
  const {
    loading,
    loaded,
    params,
    applicationInstances
  } = props;

  useEffect(() => {
    if (!loading && !loaded) {
      props.getApplicationInstance(
        params.applicationId,
        params.applicationInstanceId
      );
    }
  }, []);

  const applicationInstance = _.filter(
    applicationInstances,
    (app) => app.id === _.parseInt(params.applicationInstanceId)
  )[0];

  const getConfig = () => `LTI KEY: ${applicationInstance.lti_key || ''}\n\nLTI Secret: 
  ${applicationInstance.lti_secret || ''}\n\nConfig XML: ${applicationInstance.lti_config_xml || ''}`;

  return (
    <div>
      <div className="o-grid__item u-full">
        <Textarea
          className="c-input c-input--dark"
          labelText="LTI Config XML"
          textareaProps={{
            id: 'application_instance_lti_config_xml',
            name: 'lti_config_xml',
            rows: 25,
            readOnly: true,
            value: getConfig(),
          }}
        />
      </div>
    </div>
  );
}

export default connect(select, ApplicationInstanceActions)(XmlConfig);

XmlConfig.propTypes = {
  applicationInstance: PropTypes.shape({
    config: PropTypes.string,
    lti_config: PropTypes.string,
    lti_config_xml: PropTypes.string,
  }),
  loading: PropTypes.bool,
  loaded: PropTypes.bool,
  applicationInstances: PropTypes.array,
  getApplicationInstance: PropTypes.func,
  params: PropTypes.shape({
    applicationId: PropTypes.string.isRequired,
    applicationInstanceId: PropTypes.string.isRequired,
  }).isRequired,
  location: PropTypes.shape({
    state: PropTypes.shape({
      setNewInstance: PropTypes.func,
    }),
  }),
};
