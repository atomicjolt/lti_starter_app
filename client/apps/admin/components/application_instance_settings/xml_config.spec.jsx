import React from 'react';
import TestRenderer from 'react-test-renderer';

import { XmlConfig } from './xml_config';

describe('XmlConfig', () => {
  it('renders the XmlConfig component', () => {

    const applicationInstance = {
      id: 1,
    };

    const applicationInstances = [
      applicationInstance,
    ];

    const params = {
      applicationId: "1",
      applicationInstanceId: "1",
    };

    const result = TestRenderer.create(
        <XmlConfig applicationInstances={applicationInstances} params={params}/>
    );
    expect(result).toMatchSnapshot();
  });
});
