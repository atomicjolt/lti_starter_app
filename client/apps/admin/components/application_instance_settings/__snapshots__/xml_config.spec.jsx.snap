// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XmlConfig renders the XmlConfig component 1`] = `
<div>
  <div
    className="o-grid__item u-full"
  >
    <label
      className="c-input c-input--dark"
      htmlFor="application_instance_lti_config_xml"
    >
      <span>
        LTI Config XML
      </span>
      <textarea
        id="application_instance_lti_config_xml"
        name="lti_config_xml"
        readOnly={true}
        rows={25}
        value="LTI KEY: 

LTI Secret: 
  

Config XML: "
      />
    </label>
  </div>
</div>
`;
